doc/Makefile

```diff
@@ -21,4 +21,5 @@ clean:
 # Catch-all target: route all unknown targets to Sphinx using the new
 # "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
 %: Makefile
+	../tools/setup_tutorial_docs.sh $(SOURCEDIR)
```

Comment:
Can you move it to the main `Makefile`, and try to see if making use of target dependencies would work? Ie. similar to (in the main `Makefile`):

```
doc/place/for/tutorials:
    mkdir -p doc/place/for/tutorials (if needed)
    call_script ...

doc: doc/place/for/tutorials
    ...
```

The idea (bar existing convoluted syntax) is to be able to preserve the separation between "generating the content used by input as sphinx" (the `better-apidoc` and `sphinx-autogen` would fall in that category) and "invoking sphinx", in the end being able to say "do not regenerate the `doc/place/for/tutorials` folders if it is present" in an easy way.

It's mostly for convenience and flexibility when working with the docs: for example, being able to manually manage the tutorial files (working on a branch other than master, etc) or not take it into account for quicker sphinx building times.